src/main/java/lotto/model/Lotto.java

```diff
@@ -0,0 +1,52 @@
+package lotto.model;
+
+import java.util.HashSet;
+import java.util.List;
+
+public class Lotto {
+    private static final int LOTTO_NUMBER_COUNT = 6;
+    private static final int MIN_NUMBER = 1;
+    private static final int MAX_NUMBER = 45;
+    private static final String ERROR_INVALID_NUMBER_COUNT = "[ERROR] 로또 번호는 6개여야 합니다.";
+    private static final String ERROR_DUPLICATE_NUMBER = "[ERROR] 로또 번호는 중복될 수 없습니다.";
+    private static final String ERROR_INVALID_RANGE = "[ERROR] 로또 번호는 1에서 45 사이여야 합니다.";
+
+    private final List<Integer> numbers;
+
+    public Lotto(List<Integer> numbers) {
+        validate(numbers);
+        this.numbers = numbers;
+    }
+
+    private void validate(List<Integer> numbers) {
+        if (numbers.size() != LOTTO_NUMBER_COUNT) {
+            throw new IllegalArgumentException(ERROR_INVALID_NUMBER_COUNT);
+        }
+        if (numbers.size() != new HashSet<>(numbers).size()) {
+            throw new IllegalArgumentException(ERROR_DUPLICATE_NUMBER);
+        }
+        for (int number : numbers) {
+            if (number < MIN_NUMBER || number > MAX_NUMBER) {
+                throw new IllegalArgumentException(ERROR_INVALID_RANGE);
+            }
+        }
+    }
+
+    public List<Integer> getNumbers() {
+        return numbers;
+    }
+
+    public int countMatchingNumbers(List<Integer> winningNumbers) {
+        int matchCount = 0;
+        for (int number : numbers) {
+            if (winningNumbers.contains(number)) {
+                matchCount++;
+            }
+        }
```

Comment:
이 부분도 stream 사용하시면 좋을 것 같아요!

```suggestion
        matchCount = (int) numbers.stream()
                          .filter(winningNumbers::contains)
                          .count();
```